js/src/utils/external.ts

```diff
@@ -21,36 +21,48 @@ export async function sendProcessReq(info: {
     );
     return true;
   }
+
   try {
-    // Spawn a child process to execute a Node.js script
+    // Use node-fetch for making HTTP requests
+    const url = new URL(info.url);
     const child = spawn("node", [
       "-e",
       `
-        const http = require('http');
+        const http = require('${url.protocol === "https:" ? "https" : "http"}');
```

Comment:
For HTTPS requests, it's recommended to add SSL/TLS verification options:
```typescript
const options = {
  ...options,
  rejectUnauthorized: true, // Enforce SSL certificate validation
};
```
This ensures secure communication when making HTTPS requests.